### ISSUE TITLE
Field is unregistered when re-arranged in UI

### ISSUE THREAD

**Issue by singhba** [erikras/redux-form#2223]
```jsx
StockAppState = {
	Primary: [{name: 'ASX'}],
	Secondary: [{name: 'NASDAQ'}]
}

class StockApp extends React.Component {
	render() {
		<div>
			<Component1 Stock={this.props.StockAppState.Primary}
			<Component2 Stock={this.props.StockAppState.Secondary}
		</div>
	}
}
```
When I swap state Primary=[{name: 'NASDAQ'}] & Secondary=[{name: 'ASX'}], it unregisters NASDAQ field.

**Comment by erikras** [comment]
It's unclear to me which component is your form, and where your fields are. If you unmount a field, it will be unregistered. If you remount it again, it will be re-registered.

**Comment by huan086** [comment]
Related: https://github.com/erikras/redux-form/issues/2016

**Comment by nktssh** [comment]
possibly related #1705

**Comment by erikras** [comment]
Fix released as [`v6.5.0`](https://github.com/erikras/redux-form/releases/tag/v6.5.0).
